### ISSUE TITLE
Deemphasize seeders in peer name list UI

### ISSUE THREAD

**Issue by nikolaiwarner** [cabal-club/cabal#51]
If there is a way to tell if connected peers are seeders, it would be nice to change the color of their nick or place them at the end of the list so that real user nicks are more prominent.

**Comment by noffle** [comment]
Good idea. On the initial handshake, we could include a bit in the userdata
(where we currently pass in the peer's key) that says "I am a seeder". We could
expose that through the cabal-node API so clients can choose how to render them.

**Comment by noffle** [comment]
(BTW I would say *emphasize* seeders, since, as a non-seeder it's a relief to
know that someone is keeping the cabal alive.)

**Comment by nikolaiwarner** [comment]
Yeah! I'm not interested in seeing seeder's names though. But I really care about who's online at the moment. I think it would be really cool to just show a count of the seeders to give the user some confidence. Then in another screen, show a named list of seeders and also their uptime.

It also would be interesting if there was an indicator shown when coming back online telling you that there were no seeders around when you were gone... to give you a clue that there may be gaps in your history.

**Comment by noffle** [comment]
Showing seeders feels meaningful to me. People who dedicate hardware and network
bandwidth to helping a cabal stay alive and well resourced.. that feels
significant to me. I'd personally like to know who's putting in that effort to
keep things running. These are probably individuals running seeds, not some
nameless corporate cloud.

Maybe a compromise could be putting the peers above the seeders on the list?

**Comment by nikolaiwarner** [comment]
That's a great point! I'm with you.

What about showing an icon or symbol along with the name indicating if a peer is a seeder or regular peer?

I suppose my real goal here is inform the user if another peer has a human behind it who can respond. A set of symbols could also be useful to identify bots when we get to that 🤖

**Comment by noffle** [comment]
+1. Since we don't have roles like irc, we could use & or + or ! or some other
sigil to signify seeder.

**Comment by cblgh** [comment]
+1 to sigils/runes

**Comment by noffle** [comment]
If anyone is interested in working on this, I'd be happy to help mentor! Here's the TODOs I see:

- [ ] in cabal-cli, pass `argv.seed` into cabal-client
- [ ] in cabal-client, pass that seed variable into [`cabal.swarm()`](https://github.com/cabal-club/cabal-client/blob/master/src/client.js#L164)
- [ ] in cabal-core, edit [the initial handshake hypercore extension](https://github.com/cabal-club/cabal-core/blob/master/swarm.js#L45) to include whether the peer is a seed
- [ ] in cabal-core, edit the `_addConnection()` function to accept something like `opts.isSeed` and emit that in the `peer-added` event
- [ ] back into cabal-client, edit that event listener in `src/cabal-details.js` to set something like `user.seed = true` if it's a seeding peer
- [ ] back into cabal-cli, edit `views.js`'s `renderNicks` function to check for `user.seed` and render a sigil next to their name
- [ ] bonus: sort seeds to the bottom of the online peers list! this is in `util.js`'s `cmpUser`
